src/editorwidget/widget.ts

```diff
@@ -78,6 +108,157 @@ class EditorWidget extends CodeMirrorWidget {
       }
     });
   }
+
+  /**
+   * Creates a menu bar that is then attached as a child to layout
```

Comment:
This is too much information, perhaps "create a menu bar for the editor widget".